src/init.cc

```diff
@@ -773,11 +774,16 @@ static ncclResult_t initTransportsRank(struct ncclComm* comm, ncclUniqueId* comm
     collNetGraph.typeInter = std::min(allGather3Data[i].collNet.typeInter, collNetGraph.typeInter);
   }
 
+  // Read SCCL algorithms first, but do not connect them yet.
   int scclMinRequireNChannels = 0;
   int numValidSCCLAlgos; // We only use this at connect site
-  if (getenv("SCCL_XML_FILES")){
-    // Read SCCL algorithms first, but do not connect them yet.
-    NCCLCHECK(scclGetAllAlgoFromXMLFilesAndSetComm(comm, getenv("SCCL_XML_FILES")));
+  if (getenv("SCCL_XML_FILES") || getenv("SCCL_CONFIG")) {
+    if (getenv("SCCL_XML_FILES")){
+      NCCLCHECK(scclGetAllAlgoFromXMLFilesAndSetComm(comm, getenv("SCCL_XML_FILES")));
+    }
+    if (getenv("SCCL_CONFIG")) {
```

Comment:
what happens if both ENV are present then?

Reply:
If both are present then SCCL_XML_FILES algorithms are loaded first, so they get indices from 0 (which can be used in `ncclCustomCollective`). Only algorithms listed in SCCL_CONFIG will be considered for selection in the standard collectives.